### PR TITLE
Synchornous ADC API

### PR DESCRIPTION
**Pull request by AnneOnciulescu**
Started working on [#10](https://github.com/UPB-CS-OpenSourceUpstream/libtock-rs/issues/10). Still a draft, a lot more to be done.

### CODE REVIEW

**Review by alexandruradovici — COMMENTED** [review]
I would start only by providing the simple read API, that reads one single value.

**Comment by alexandruradovici** [comment]
@AnneOnciulescu any progress?

File: apis/adc/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            sample.set(Some(adc_val));
+        });
+        share::scope(|subscribe| {
+            if let Ok(()) = Self::register_listener(&listener, subscribe) {
+                if let Ok(()) = Self::read_single_sample() {
+                    while sample.get().is_none() {
+                        S::yield_wait();
+                    }
+                }
+            }
```
Comment (alexandruradovici) [inline apis/adc/src/lib.rs:48]: Use the ?, similar to https://github.com/DanutAldea/libtock-rs/blob/b70eb0cbe1766a37585472235e77f755a7c12786/apis/ambient_light/src/lib.rs#L45-L46

**Review by alexandruradovici — CHANGES_REQUESTED** [review]
Looks good, just make the changes and let's send the first version to upstream.

File: examples/adc.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+stack_size! {0x200}
+
+fn main() {
+    match Adc::exists() {
+        Ok(()) => writeln!(Console::writer(), "adc driver available").unwrap(),
+        Err(_) => {
+            writeln!(Console::writer(), "adc driver unavailable").unwrap();
+            return;
+        }
+    }
```
Comment (alexandruradovici) [inline examples/adc.rs:24]:
```suggestion
    if Adc::exists().is_err() {
        writeln!(Console::writer(), "adc driver unavailable").unwrap();
        return;
    }
```

**Review by alexandruradovici — DISMISSED** [review]
Looks good, a small fix in the sample and send it to upstream.

File: apis/adc/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                None => Err(ErrorCode::Busy),
+                Some(adc_val) => Ok(adc_val),
+            }
+        })
+    }
+
+    // pub fn sample_continuous()
+    // pub fn sample_buffer()
+    // pub fn sample_buffer_continuous()
+    // pub fn stop_sampling()
```
Comment (alexandruradovici) [inline apis/adc/src/lib.rs:59]: Delete this.

**Review by alexandruradovici — DISMISSED** [review]

**Comment by AnneOnciulescu** [comment]
[tock#471](https://github.com/tock/libtock-rs/pull/471)

**Pull request merged**